### PR TITLE
fix: support avg queries

### PR DESCRIPTION
**Pull request by TheSlimvReal**
Actually, I am not sure if something like sum of all actual divided by sum of all totals is more required. Will check.

### CODE REVIEW

File: src/app/core/export/query.service.ts [hunk 1]
```diff
@@ -248,6 +249,25 @@ export class QueryService {
...
+   * @private
+   */
+  private avg(data: any[], decimals = 0): string {
+    const numbers = data
+      .map((d) => Number.parseInt(d))
+      .filter((i) => !Number.isNaN(i));
+    const result =
+      numbers.length === 0
+        ? 0
+        : numbers.reduce((i, sum) => sum + i, 0) / numbers.length;
```
Comment (TheSlimvReal) [inline src/app/core/export/query.service.ts:267]: Should the avg be calculated over all values or just the valid/existing ones?

File: src/app/core/export/query.service.spec.ts [hunk 2]
```diff
@@ -635,6 +635,30 @@ describe("QueryService", () => {
...
+    // Returns valid number if no values are available
+    res = service.queryData("a:avg", undefined, undefined, [{ b: 1 }]);
+    expect(res).toBe("0");
+
+    // Numbers are fixed to provided decimals
+    res = service.queryData("a:avg(2)", undefined, undefined, [
+      { a: 3 },
+      { a: 2 },
+    ]);
+    expect(res).toBe("2.50");
```
Comment (sleidig) [inline src/app/core/export/query.service.spec.ts:659]: let's add a test with one entry `= 0` (I agree that missing and invalid should be ignored, but 0 should be included in calculation)

File: src/app/core/export/query.service.ts (hunk 1 above)
Comment (sleidig) [reply src/app/core/export/query.service.ts:267]: I would ignore missing / invalid. (that might help if we have a mix of entity types or badly maintained data)

**Review by sleidig — APPROVED** [review]
Looks good :-)

**Pull request merged**